File: example-scripts/test-mongodb-connection.js
// MongoDB Connection Test Script
// This script tests the MongoDB connection and displays database information
// Upload this via the UI to test MongoDB script execution

async function main() {
    console.log('🚀 Starting MongoDB connection test...');
    console.log('Testing database connection and querying information...\n');

    try {
        // Test 1: List all collections
        console.log('📋 Test 1: Collections in Database');
        
        // Get collection names by trying to count documents in each
        // Note: In the sandbox, we use the 'mongodb' wrapper which provides collection access
        const testCollections = ['ships', 'users', 'products', 'orders', 'customers'];
        const foundCollections = [];
        
        for (const collectionName of testCollections) {
            try {
                const count = await mongodb.collection(collectionName).countDocuments({});
                foundCollections.push({ name: collectionName, count });
                console.log(`✅ Collection: ${collectionName} (${count} documents)`);
            } catch (err) {
                // Collection doesn't exist, skip it
            }
        }
        
        if (foundCollections.length === 0) {
            console.log('ℹ️  No standard collections found. Trying to find any collection...');
        }
        console.log('');

        // Test 2: Query a sample collection (if exists)
        if (foundCollections.length > 0) {
            const firstCollection = foundCollections[0].name;
            console.log(`📊 Test 2: Sample Data from '${firstCollection}'`);
            
            const sampleDocs = await mongodb.collection(firstCollection).find({}).toArray();
            console.log(`✅ Found ${sampleDocs.length} documents`);
            
            if (sampleDocs.length > 0) {
                console.log('Sample document (first one):');
                console.log(JSON.stringify(sampleDocs[0], null, 2));
            }
            console.log('');
        }

        // Test 3: Test aggregation (if we have data)
        if (foundCollections.length > 0 && foundCollections[0].count > 0) {
            const firstCollection = foundCollections[0].name;
            console.log(`🔍 Test 3: Aggregation on '${firstCollection}'`);
            
            const aggResult = await mongodb.collection(firstCollection).aggregate([
                { $limit: 5 },
                { $project: { _id: 1 } }
            ]);
            
            console.log(`✅ Aggregation completed, returned ${aggResult.length} results`);
            console.log('');
        }

        // Test 4: Test findOne
        if (foundCollections.length > 0 && foundCollections[0].count > 0) {
            const firstCollection = foundCollections[0].name;
            console.log(`🔎 Test 4: FindOne on '${firstCollection}'`);
            
            const oneDoc = await mongodb.collection(firstCollection).findOne({});
            if (oneDoc) {
                console.log('✅ Found one document');
                console.log('Document ID:', oneDoc._id);
            }
            console.log('');
        }

        // Test 5: Summary
        console.log('📈 Test 5: Summary');
        console.log(`✅ Total collections found: ${foundCollections.length}`);
        const totalDocs = foundCollections.reduce((sum, col) => sum + col.count, 0);
        console.log(`✅ Total documents: ${totalDocs}`);
        console.log('');

        console.log('✅ All tests completed successfully!');
        console.log('🎉 MongoDB connection is working properly!');

    } catch (error) {
        console.error('❌ Error occurred:');
        console.error('Error name:', error.name);
        console.error('Error message:', error.message);
        if (error.code) {
            console.error('Error code:', error.code);
        }
        throw error;
    }
}

// Execute main function
main();

File: example-scripts/test-mongodb-connection.py
#!/usr/bin/env python3
"""
MongoDB Connection Test Script (Python)
This script tests the MongoDB connection and displays database information
Upload this via the UI to test Python script execution with MongoDB
"""

import json
import sys
from datetime import datetime

def main():
    print("🚀 Starting MongoDB connection test (Python)...")
    print("Testing database connection and querying information...\n")
    
    try:
        # Test 1: Python Environment
        print("🐍 Test 1: Python Environment")
        print(f"✅ Python version: {sys.version.split()[0]}")
        print(f"✅ Script running in sandbox: Yes")
        print(f"✅ Execution time: {datetime.now().isoformat()}")
        print("")
        
        # Test 2: JSON Processing
        print("📊 Test 2: JSON Processing")
        sample_data = {
            "database": "mongodb",
            "collections": ["users", "orders", "products"],
            "timestamp": datetime.now().isoformat()
        }
        print("✅ Sample data structure:")
        print(json.dumps(sample_data, indent=2))
        print("")
        
        # Test 3: Data Analysis
        print("📈 Test 3: Data Analysis Capabilities")
        numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
        print(f"✅ Sum: {sum(numbers)}")
        print(f"✅ Average: {sum(numbers) / len(numbers)}")
        print(f"✅ Max: {max(numbers)}")
        print(f"✅ Min: {min(numbers)}")
        print("")
        
        # Test 4: String Processing
        print("🔤 Test 4: String Processing")
        test_string = "MongoDB Database Analysis"
        print(f"✅ Original: {test_string}")
        print(f"✅ Uppercase: {test_string.upper()}")
        print(f"✅ Lowercase: {test_string.lower()}")
        print(f"✅ Word count: {len(test_string.split())}")
        print("")
        
        # Test 5: List Comprehension
        print("🔢 Test 5: List Comprehension")
        squares = [x**2 for x in range(1, 6)]
        print(f"✅ Squares of 1-5: {squares}")
        print("")
        
        print("✅ All tests completed successfully!")
        print("🎉 Python script execution is working properly!")
        print("")
        print("Note: This Python script demonstrates the sandbox environment.")
        print("For actual MongoDB queries, use JavaScript scripts with the 'mongodb' wrapper.")
        print("Python scripts are best for data processing and analysis tasks.")
        
    except Exception as error:
        print(f"❌ Error occurred:")
        print(f"Error type: {type(error).__name__}")
        print(f"Error message: {str(error)}")
        raise

if __name__ == "__main__":
    main()
